subcommands/keys/tuf_updates_init.go

```diff
@@ -0,0 +1,58 @@
+package keys
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+func init() {
+	initCmd := &cobra.Command{
+		Use:   "init",
+		Short: "Start a new transaction to update TUF root keys",
+		Run:   doTufUpdatesInit,
+	}
+	initCmd.Flags().StringP("changelog", "m", "", "Reason for doing this operation. Saved in root metadata to track change history.")
+	_ = initCmd.MarkFlagRequired("changelog")
+	initCmd.Flags().BoolP("first-time", "", false, "Used for the first customer rotation. The command will download the initial root key.")
+	initCmd.Flags().StringP("keys", "k", "", "Path to <offline-creds.tgz> used to store initial root key.")
+	_ = initCmd.MarkFlagFilename("keys")
+	initCmd.MarkFlagsRequiredTogether("first-time", "keys")
+	tufUpdatesCmd.AddCommand(initCmd)
+}
+
+func doTufUpdatesInit(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	changelog, _ := cmd.Flags().GetString("changelog")
+	firstTime, _ := cmd.Flags().GetBool("first-time")
+	keysFile, _ := cmd.Flags().GetString("keys")
+
+	if firstTime {
+		if _, err := os.Stat(keysFile); err == nil {
+			subcommands.DieNotNil(errors.New(`Destination file exists.
+Please make sure you aren't accidentally overwriting another factory's keys`))
+		}
+		subcommands.AssertWritable(keysFile)
+	}
+
+	res, err := api.TufRootUpdatesInit(factory, changelog, firstTime)
+	subcommands.DieNotNil(err)
+
+	fmt.Printf(`A new transaction to update TUF root keys started.
+Your transaction ID is %s.
+Please, keep it secret and only share with participants of the transaction.
+Nobody can make changes to the transaction without this ID other than cancel it.`,
+		res.TransactionId)
+
+	if firstTime {
+		creds := make(OfflineCreds)
+		creds["tufrepo/keys/first-root.sec"] = []byte(res.FirstRootKeyPriv)
+		creds["tufrepo/keys/first-root.pub"] = []byte(res.FirstRootKeyPub)
+		saveTufCreds(keysFile, creds)
```

Comment:
What happens if the request `api.TufRootUpdatesInit(factory, changelog, firstTime)` succeeds but `saveTufCreds` fails? Is it possible to run the command with `first-time` again?

Reply:
Yes, the user will see the error... So they can cancel the failed TUF updates and start anew with the `--first-time` option again.

Reply:
Will the backend permit running the `first time` for the second time; the backend thinks that the first attempt was successful?

Reply:
The backend is indifferent to how many times the user runs `--first-time` as long as the new TUF root was not yet applied.

Reply:
cool, that's what I thought.